src/oracle/query_builder/limit_clause.rs

```diff
@@ -0,0 +1,19 @@
+/// TODO: this is currently not used, IDK why diesel still takes the original limit clause
```

Comment:
Diesel just don't know about this type. I think we could make this (somehow) work.

Short version:
We need to define our own [limit function](https://github.com/diesel-rs/diesel/blob/master/diesel/src/query_dsl/mod.rs#L729) that returns our own type. The best way to do this is to define a new trait and implement that for all T that implement `QueryDsl`. After that we could use this (new) function everywhere where we explicitly call limit. The only place where this won't work is in the implementation of `first` because that is inside of diesel itself.

Reply:
Once specialization lands you should also just be able to specialize the original impl. We can already do that today if we're ok with making this crate nightly only.

Reply:
@weiznich I am not sure what you mean. We might discuss that via chat or in person.